Move answer document

sairamya93 opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What

To handle the rendering of Answer document type from government_frontend to frontend as part of [RFC 175]

## Why

Trello card https://trello.com/c/Dv4uSqX0/525-move-answer-from-government-frontend-to-frontend, [Jira issue PNP-6415](https://gov-uk.atlassian.net/browse/PNP-6415)

## How

To add a scoped route for a controller and a view with the required functionalities to get Answer document working.
There was no need for a model and a presenter for now.

**Note:** Answer is part of tests for `document_collection_presenter` and `step_navigation_controller_test` but are not added in frontend for now.

https://github.com/alphagov/government-frontend/blob/a78c77bc0756f667b041cac225b5e13cb34267a1/app/presenters/document_collection_presenter.rb#L61
https://github.com/alphagov/government-frontend/blob/a78c77bc0756f667b041cac225b5e13cb34267a1/test/controllers/step_navigation_controller_test.rb#L6

## Screenshots?

| Government-frontend | Frontend |
|-------|--------|
|![image](https://github.com/user-attachments/assets/3563a039-09ea-4a4f-837b-755ae1c24971)| ![image](https://github.com/user-attachments/assets/1590554e-0777-49ed-9488-b3d7307417ca) |
| ![image](https://github.com/user-attachments/assets/504e396a-d48f-41db-94ae-0a45879d3a68) | ![image](https://github.com/user-attachments/assets/1f830698-bd5e-45ec-a722-aa6ef9ebe2c1) |
| ![image](https://github.com/user-attachments/assets/e27c8e3f-fe25-4875-acdc-4ff8fae9aa6d) | ![image](https://github.com/user-attachments/assets/c93cf133-c844-497f-8c8c-e7ddeb50959c) |
| ![image](https://github.com/user-attachments/assets/e3edc6de-6bc8-432a-93d9-fee2fab2e974) | ![image](https://github.com/user-attachments/assets/7d2ca7d5-9c37-4582-bd7c-7f0728c5729b) |

sairamya93 committed:
Update README.md and govuk_examples.yml

sairamya93 committed:
Add request tests for Answer

sairamya93 committed:
Add system tests for Answer

- Added additional test to verify an Answer page in English
- Answer presenter tests contain tests for title and body which is already part of the content_item in frontend and verified in system tests

Audit trail for tests: https://github.com/alphagov/government-frontend/blob/340bb88f3f8ff9e15281020ce2850e2be99a8b8b/test/integration/answer_test.rb
Audit trail for FAQ helper method: https://github.com/alphagov/government-frontend/blob/340bb88f3f8ff9e15281020ce2850e2be99a8b8b/test/test_helper.rb#L265
Audit trail for answer presenter test: https://github.com/alphagov/government-frontend/blob/a78c77bc0756f667b041cac225b5e13cb34267a1/test/presenters/answer_presenter_test.rb

deborahchua reviewed (commented):
This looks good! I've left a few small comments

sairamya93 commented on spec/support/faq_page_helpers.rb:
```diff
@@ -0,0 +1,8 @@
+module FaqPageHelper
```
Oh yes that's right! I missed it.
I have now updated the test with the existing support method - `find_schema_of_type`

sairamya93 committed:
Update view for Answer

- Copied over the view from government-frontend
Audit trail: https://github.com/alphagov/government-frontend/blob/340bb88f3f8ff9e15281020ce2850e2be99a8b8b/app/views/content_items/answer.html.erb

sairamya93 committed:
Remove requested_variant in View

- The requested_variant is removed in frontend but could be added later when required to test

sairamya93 committed:
Add body_with_related_links content in View

- The rendering of content_items/body_with_related_links is directly replaced with the content itself as it is being referenced only in Answer document view.
Also simple_header in the content_for tag is removed as it is not necessary

Audit trail for body_with_related_links: https://github.com/alphagov/government-frontend/blob/a6565d7e6ffd323a544ebf38c55fbebf1aa7cc9c/app/views/content_items/_body_with_related_links.html.erb
Ref link that mentions simple_header removed in few of the document types like document_collection: https://github.com/alphagov/government-frontend/pull/837/commits/d5b5e93172f90c98df76fcb12d5acb92f80a0fcb

KludgeKML commented on app/controllers/answer_controller.rb:
```diff
@@ -0,0 +1,7 @@
+  include Cacheable
+
+  def show
+    @presenter = ContentItemPresenter.new(content_item)
```
Might want to move this line to a later commit, since we're not using the presenter yet?

sairamya93 replied:
Ah yes updated in the commit where it is used in the view

KludgeKML reviewed (changes requested):
Looking good, one note about ordering.

sairamya93 committed:
Add answer route and controller

sairamya93 committed:
Add page title in View

- Added page_title,description and replaced 'extra_head_content' with 'extra_headers'

KludgeKML reviewed (approved):
LGTM!
